gradle: update to 8.14.2

RawDiamondMC opened the pull request:
Topic Description
-----------------

- gradle: update to 8.14.1

Package(s) Affected
-------------------
- gradle: 8.14.1

Security Update?
----------------

No

Build Order
-----------

```
#buildit gradle
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] Architecture-independent `noarch`

RawDiamondMC commented:
@aosc-buildit-bot build

RawDiamondMC commented:
@aosc-buildit-bot build

RawDiamondMC commented:
@aosc-buildit-bot build

RawDiamondMC commented:
@aosc-buildit-bot build

RawDiamondMC commented:
Basic commands are tested.
Initialize a project:
```
rawdiamondmc@aosc [ test ] $ gradle init
NOTE: Picked up JDK_JAVA_OPTIONS:  --module-path=/usr/share/openjfx/lib/

Welcome to Gradle 8.14.1!

Here are the highlights of this release:
 - Java 24 support
 - GraalVM Native Image toolchain selection
 - Enhancements to test reporting
 - Build Authoring improvements

For more details see https://docs.gradle.org/8.14.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

Found existing files in the project directory: '/home/rawdiamondmc/test'.
Directory will be modified and existing files may be overwritten.  Continue? (default: no) [yes, no] yes

Select type of build to generate:
  1: Application
  2: Library
  3: Gradle plugin
  4: Basic (build structure only)
Enter selection (default: Application) [1..4] 1

Select implementation language:
  1: Java
  2: Kotlin
  3: Groovy
  4: Scala
  5: C++
  6: Swift
Enter selection (default: Java) [1..6] 1

Enter target Java version (min: 7, default: 21): 

Project name (default: test): 

Select application structure:
  1: Single application project
  2: Application and library project
Enter selection (default: Single application project) [1..2] 2

Select build script DSL:
  1: Kotlin
  2: Groovy
Enter selection (default: Kotlin) [1..2]  

Generate build using new APIs and behavior (some features may change in the next minor release)? (default: no) [yes, no] yes


> Task :init
Learn more about Gradle by exploring our Samples at https://docs.gradle.org/8.14.1/samples/sample_building_java_applications_multi_project.html

BUILD SUCCESSFUL in 1m 24s
1 actionable task: 1 executed
```
First build:
```
rawdiamondmc@aosc [ test ] $ gradle build
NOTE: Picked up JDK_JAVA_OPTIONS:  --module-path=/usr/share/openjfx/lib/
Calculating task graph as no cached configuration is available for tasks: build

> Task :list:test
NOTE: Picked up JDK_JAVA_OPTIONS:  --module-path=/usr/share/openjfx/lib/

> Task :app:test
NOTE: Picked up JDK_JAVA_OPTIONS:  --module-path=/usr/share/openjfx/lib/

BUILD SUCCESSFUL in 5m 19s
22 actionable tasks: 22 executed
Configuration cache entry stored.
```
Second build:
```
rawdiamondmc@aosc [ test ] $ gradle build
NOTE: Picked up JDK_JAVA_OPTIONS:  --module-path=/usr/share/openjfx/lib/
Reusing configuration cache.

BUILD SUCCESSFUL in 4s
13 actionable tasks: 13 up-to-date
Configuration cache entry reused.
```

MingcongBai commented on lang-java/gradle/autobuild/defines:
```diff
@@ -1,6 +1,6 @@
 PKGNAME=gradle
 PKGSEC=java
-PKGDEP="openjdk"
+PKGDEP="openjdk-17 openjdk-11"
```
This dependency makes no sense.

xtexx commented on lang-java/gradle/autobuild/build:
```diff
@@ -1,17 +1,27 @@
+echo org.gradle.java.installations.fromEnv=JDK11,JDK17 >> gradle.properties
+
+abinfo "Building gradle ..."
+JDK11=/lib/java-11 JDK17=/lib/java-17 ./gradlew installAll -Pgradle_installPath="$SRCDIR"/temp -PfinalRelease=true --no-configuration-cache
```
Why does it need two JDKs? How about using the minimal supported JDK version to build it.

xtexx replied:
`--no-daemon`

xtexx commented on lang-java/gradle/autobuild/build:
```diff
@@ -1,17 +1,25 @@
+install -dv "$SRCDIR"/temp
+
+abinfo "Building gradle ..."
+./gradlew installAll -Pgradle_installPath="$SRCDIR"/temp -PfinalRelease=true --no-configuration-cache --no-daemon
```
Possibly bike-shedding, I prefer to explicitly specify the JDK version to use, or else it is possible for a wrong OpenJDK version (which is transitively included into BuildKit) to be used (and lead to incorrect class file versions).

RawDiamondMC replied:
> Possibly bike-shedding, I prefer to explicitly specify the JDK version to use, or else it is possible for a wrong OpenJDK version (which is transitively included into BuildKit) to be used (and lead to incorrect class file versions).

https://github.com/gradle/gradle/blob/v8.14.2/settings.gradle.kts#L330

If the java is not correct, the build will fail.

xtexx replied:
ah. didn't notice that.

MingcongBai commented on lang-java/gradle/autobuild/build:
```diff
@@ -1,17 +1,25 @@
+install -dv "$SRCDIR"/temp
+
+abinfo "Building gradle ..."
+./gradlew installAll -Pgradle_installPath="$SRCDIR"/temp -PfinalRelease=true --no-configuration-cache --no-daemon
```
Should be "$SRCDIR"/gradlew

RawDiamondMC replied:
fixed

RawDiamondMC committed:
gradle: update to 8.14.2

- revert fb28bd4 and build gradle from source
- pin the version of openjdk to openjdk-17
- exclude the bat file in the bin folder
